poshc2/server/Implant.py

```diff
@@ -71,23 +71,46 @@ def display(self):
         try:
             Pushover_APIToken = select_item("Pushover_APIToken", "C2Server")
             Pushover_APIUser = select_item("Pushover_APIUser", "C2Server")
-
-            if EnableNotifications.lower().strip() == "yes":
+            if EnableNotifications.lower().strip() == "yes" and Pushover_APIToken not in ("", None):
                 conn = http.client.HTTPSConnection("api.pushover.net:443")
                 conn.request("POST", "/1/messages.json",
-                             urllib.parse.urlencode({
-                                 "token": Pushover_APIToken,
-                                 "user": Pushover_APIUser,
-                                 "message": "[%s] - NewImplant: %s @ %s" % (NotificationsProjectName, self.User, self.Hostname),
-                             }), {"Content-type": "application/x-www-form-urlencoded"})
-
+                                 urllib.parse.urlencode({
+                                     "token": Pushover_APIToken,
+                                     "user": Pushover_APIUser,
+                                     "message": "[%s] - NewImplant: %s @ %s" % (NotificationsProjectName, self.User, self.Hostname),
+                                 }), {"Content-type": "application/x-www-form-urlencoded"})
                 output = conn.getresponse()
                 if output.status != 200:
                     data = output.read()
                     print("\nPushover error: ")
                     print(data)
         except Exception as e:
             print("Pushover send error: %s" % e)
+        try:
+            Slack_BotToken = select_item("Slack_BotToken", "C2Server")
+            if EnableNotifications.lower().strip() == "yes" and Slack_BotToken not in ("", None):
```

Comment:
this can just be `and Slack_BotToken` which will check if it's a null or empty string :)

Reply:
I'll sort this out tonight and update the PR

Reply:
You don't have to it'll work fine it's more just an FYI !